File: apps/mobile/targets/positions-widget/Sources/Utilities/WidgetDeepLink.swift
import Foundation

enum WidgetDeepLink {
  static let scheme = "depthly"

  static func forPosition(ref: String) -> URL? {
    let encoded = ref.addingPercentEncoding(withAllowedCharacters: .alphanumerics) ?? ref
    return URL(string: "\(scheme):///positions/\(encoded)")
  }
}

File: apps/mobile/modules/widget-bridge/android/src/main/java/expo/modules/widgetbridge/widget/deeplink/WidgetDeepLink.kt
package expo.modules.widgetbridge.widget.deeplink

import android.content.Context
import android.content.Intent
import android.net.Uri

object WidgetDeepLink {
  const val SCHEME = "depthly"

  fun openPositionIntent(context: Context, ref: String): Intent {
    val encoded = Uri.encode(ref)
    return Intent(Intent.ACTION_VIEW, Uri.parse("$SCHEME:///positions/$encoded")).apply {
      addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TOP)
    }
  }
}
